packages/twenty-server/src/engine/core-modules/open-api/utils/components.utils.ts

```diff
@@ -31,28 +62,55 @@ const getFieldProperties = (type: FieldMetadataType): Property => {
     case FieldMetadataType.EMAIL:
       return { type: 'string', format: 'email' };
     case FieldMetadataType.DATE_TIME:
+      return { type: 'string', format: 'date-time' };
     case FieldMetadataType.DATE:
       return { type: 'string', format: 'date' };
     case FieldMetadataType.NUMBER:
       return { type: 'integer' };
-    case FieldMetadataType.NUMERIC:
     case FieldMetadataType.RATING:
+      return {
+        type: 'string',
+        enum: options?.map((option: { value: string }) => option.value),
+      };
+    case FieldMetadataType.NUMERIC:
     case FieldMetadataType.POSITION:
       return { type: 'number' };
     case FieldMetadataType.BOOLEAN:
       return { type: 'boolean' };
     case FieldMetadataType.RAW_JSON:
+      if (propertyName === 'secondaryLinks') {
```

Comment:
Mmm can't we define a generic output here? I feel like this will be hard to maintain

Reply:
This openapi IS hard to maintain 😅
We miss some information in composite fields and metadata to avoid adding those patches
We may need to deep work on that but not sure it is a prio for now